### PR TITLE
Issue #1947: Nested less joins

### PR DESCRIPTION
**Pull request by hawkfish**
Fix typo in forwarding templates.
Fix issue where `SelectionVector`s were being
incrementally built out of order
and overwritten during scattering.

### CODE REVIEW

File: src/common/row_operations/row_match.cpp [hunk 1]
```diff
@@ -47,13 +47,13 @@ idx_t SelectComparison<GreaterThanEquals>(Vector &left, Vector &right, const Sel
 template <>
 idx_t SelectComparison<LessThan>(Vector &left, Vector &right, const SelectionVector *sel, idx_t count,
                                  SelectionVector *true_sel, SelectionVector *false_sel) {
-	return VectorOperations::GreaterThanEquals(left, right, sel, count, true_sel, false_sel);
+	return VectorOperations::LessThan(left, right, sel, count, true_sel, false_sel);
```
Comment (Mytherin) [inline src/common/row_operations/row_match.cpp:50]: Whoops!

**Review by Mytherin — APPROVED** [review]

**Comment by Mytherin** [comment]
Looks great, thanks for the PR!

File: src/common/row_operations/row_match.cpp (hunk 1 above)
Comment (hawkfish) [reply src/common/row_operations/row_match.cpp:50]: The funny part is that I saw the same issue somewhere else a while and fixed it, but now it's my turn!

**Comment by Mytherin** [comment]
Test failure is unrelated to this PR (HTTP error)

**Pull request merged**